### PR TITLE
various

### PR DESCRIPTION
**Pull request by basisbit**
### Before you begin

* I understand my contributions may be rejected for any reason
* I understand my contributions are for the benefit of this imageboard software
* I understand my contributions are licensed under the GNU AGPLv3

- [x] I understand all of the above

---

- violation rule titles adjusted to new sample ruleset
- updated dependencies
Cherry picked commits form other forks:
- added missing sources url validation
- fixed tag change error handling by adding a transaction as wrapper

Kudos to @liamwhite (author of cherry-picked commits) for the continued work and improvements of philomena!

### CODE REVIEW

File: assets/package-lock.json [hunk 1]
```diff
@@ -4807,8 +4807,8 @@
     },
     "lodash": {
       "version": "4.17.15",
-      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.15.tgz",
-      "integrity": "sha512-8xOcRHvCjnocdS5cpwXQXVzmmh5e5+saE2QGoeQmbKmRS6J3VQppPOIt0MnmE+4xlZoumy0GPG0D0MVIQbNA1A==",
+      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.19.tgz",
```
Comment (cultpony) [inline assets/package-lock.json:4810]: Probably best to move updates and linting fixes into separate merge requests

File: lib/philomena_web/views/report_view.ex [hunk 2]
```diff
@@ -13,17 +13,21 @@ defmodule PhilomenaWeb.ReportView do
...
+      "Rule 2a: Unrelated to MLP": "Rule 2a",
+      "Rule 2b: Hateful content": "Rule 2b",
+      "Rule 2c: Underage porn": "Rule 2c",
+      "Rule 2d: Illegal content": "Rule 2d",
+      "Rule 2e: Banned topic": "Rule 2e",
+      "Rule 3a: Forcing opinion": "Rule 3a",
+      "Rule 3b: NSFW content in non-filterable medium": "Rule 3b",
+      "Rule 3c: Mistagging": "Rule 3c",
+      "Rule 4a: Incorrect sourcing": "Rule 4a",
+      "Rule 4b: Breach of DNP": "Rule 4b",
```
Comment (cultpony) [inline lib/philomena_web/views/report_view.ex:30]: Do we have consensus on those rules?

File: assets/package-lock.json (hunk 1 above)
Comment (basisbit) [reply assets/package-lock.json:4810]: okay, will do so next time!

File: lib/philomena_web/views/report_view.ex (hunk 2 above)
Comment (basisbit) [reply lib/philomena_web/views/report_view.ex:30]: that was the first voting draft. So far people only voted for it and no one against it.

**Pull request merged**